lib/modules/datasource/rubygems/metadata-cache.ts

```diff
@@ -21,44 +32,61 @@ export class MetadataCache {
   ): Promise<ReleaseResult> {
     const cacheNs = `datasource-rubygems`;
     const cacheKey = `metadata-cache:${registryUrl}:${packageName}`;
-    const hash = toSha256(versions.join(''));
+    const versionsHash = hashVersions(versions);
 
-    const loadCache = (): AsyncResult<ReleaseResult, NonNullable<unknown>> =>
-      Result.wrapNullable(
+    const loadCache = (): AsyncResult<ReleaseResult, CacheError> =>
+      Result.wrapNullable<CacheRecord, CacheError, CacheError>(
         packageCache.get<CacheRecord>(cacheNs, cacheKey),
-        'cache-not-found' as const
+        'cache-not-found'
       ).transform((cache) => {
-        return hash === cache.hash
+        return versionsHash === cache.hash
           ? Result.ok(cache.data)
-          : Result.err('cache-outdated' as const);
+          : Result.err('cache-outdated');
       });
 
-    const saveCache = async (data: ReleaseResult): Promise<ReleaseResult> => {
+    const saveCache = async (
+      hash: string,
+      data: ReleaseResult,
+      ttlMinutes = 100 * 24 * 60,
+      ttlDelta = 10 * 24 * 60
```

Comment:
why as parameter? those params are never passed anywhere, so they should be const.